invenio_app_rdm/utils/files.py

```diff
@@ -0,0 +1,145 @@
+    "record_html": "{base}/records/{id}",
+}
+
+
+def ui_link_for(tpl, **kwargs):
+    """Create an UI link using specific template."""
+    UI_URL = current_app.config.get("SITE_UI_URL")
+    return link_for(UI_URL, tpl, **kwargs)
+
+
+def link_for(base, tpl, **kwargs):
+    """Create a link using specific template."""
+    tpl = URLS.get(tpl)
+    for k in [
+        "key",
+    ]:
+        if k in kwargs:
+            kwargs[k] = quote(kwargs[k].encode("utf8"))
+    return tpl.format(base=base, **kwargs)
+
+
+def format_file_integrity_report(report):
+    """Format the email body for the file integrity report."""
+    lines = []
+    for entry in report:
+        f = entry["file"]
+        lines.append("ID: {}".format(str(f.id)))
+        lines.append("URI: {}".format(f.uri))
+        lines.append("Name: {}".format(entry.get("filename")))
+        lines.append("Created: {}".format(f.created))
+        lines.append("Checksum: {}".format(f.checksum))
+        lines.append("Last Check: {}".format(f.last_check_at))
+        if "record" in entry:
+            lines.append(
+                "Record: {}".format(
+                    ui_link_for("record_html", id=entry["record"].get("id"))
+                )
+            )
+        if "deposit" in entry:
+            lines.append(
+                "Deposit: {}".format(
+                    ui_link_for("deposit_html", id=entry["deposit"].get("id"))
+                )
+            )
+        lines.append(("-" * 80) + "\n")
+    return "\n".join(lines)
+
+
+def generate_file_integrity_report(files):
+    """Generates a file integrity report."""
+    report = []
+    for file in files:
+        entry = _generate_report_entry(file)
+        report.append(entry)
+    return report
+
+
+def _generate_report_entry(file):
+    """Generates a single entry for the files integrity report."""
+    entry = {"file": file}
+    for o in file.objects:
+        entry["filename"] = o.key
+
+        # Find records/deposits for the files
+        deposit = get_deposit_from_bucket(o.bucket_id)
+        if deposit and deposit.json:
+            entry["deposit"] = deposit.json
+        else:
+            record = get_record_from_bucket(o.bucket_id)
+            if record and record.json:
+                entry["record"] = record.json
+    return entry
+
+
+def get_record_from_bucket(bucket_id):
+    """Retrieve a record from a bucket id."""
+    return RDMRecordMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def get_deposit_from_bucket(bucket_id):
+    """Retrieve a deposit from a bucket id."""
+    return RDMDraftMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def send_integrity_report_email(report):
+    """Sends an e-mail with the given report.
+
+    APP configs ('SECURITY_EMAIL_SENDER', 'ADMIN_EMAIL') must be set.
+    Oherwise a warning is logged.
+    """
+    try:
+        if report:
+            subject = "Zenodo files integrity report [{}]".format(datetime.now())
```

Comment:
```suggestion
            subject = "Zenodo files integrity report [{}]".format(datetime.now())
```
I think we need this a bit more customisable. It is not only Zenodo anymore. Maybe it makes sense using jinja instead which makes the email easily overridable ?

Reply:
I agree it should be customizable.

I will give it a try with Jinja, however I will need to adapt the test as well.